encoding/codecs/h264.md

```diff
@@ -0,0 +1,247 @@
+# Encoding with x264
+
+H.264 has been the de facto standard video codec
+across the internet for the past decade.
+It is widely supported for playback in all modern browsers
+and many hardware devices such as gaming consoles.
+It provides better video quality at smaller file sizes
+compared to its predecessors.
+
+x264 is a mature, free, open-source encoder for the H.264 codec.
+This guide will focus on x264.
+
+## Prerequisites
+
+To get started, you'll need two things:
+
+- A video to encode--for the examples,
+we will pipe in a video from VapourSynth,
+which you should be able to do if you've been following
+the previous sections of this guide
+- The x264 encoder
+
+Here's how we get a copy of the x264 encoder:
+
+### Windows
+
+TODO
+
+### Linux/macOS
+
+Generally, x264 will be available
+through your distribution's package manager.
+Here are a few examples:
+
+#### Ubuntu/Debian
+
+`sudo apt-get install x264`
+
+#### Arch Linux
+
+`sudo pacman -S x264`
+
+#### macOS
+
+`brew install x264`
+
+## Getting Started
+
+x264 is very configurable, and the options may seem overwhelming.
+But you can get started encoding by using the presets x264 provides
+and understanding a few basic concepts.
+We'll walk through those concepts with the following examples.
+
+## Example 1
+
+Go ahead and open up a terminal window,
+and navigate to the folder where your VapourSynth script lives.
+Let's run the following command:
+
+`vspipe --y4m myvideo.vpy - | x264 --demuxer y4m --preset veryfast --tune animation --crf 24 -o x264output.mkv -`
+
+Let's run through what each of these options means:
+
+#### `vspipe --y4m myvideo.vpy -`
```

Comment:
… but I'm not sure headings are the correct way to style this. Will probably have to see how this ends up looking to make a final call.

Reply:
![image](https://user-images.githubusercontent.com/10972040/66434523-be8f5c80-e9f0-11e9-8e9e-e6e895fd85de.png)
![image](https://user-images.githubusercontent.com/10972040/66434561-d1099600-e9f0-11e9-8a08-9817812a71cd.png)